core/trino-main/src/test/java/io/trino/sql/query/TestListagg.java

```diff
@@ -0,0 +1,301 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.query;
+
+import io.trino.spi.TrinoException;
+import io.trino.sql.parser.ParsingException;
+import org.testcontainers.shaded.org.apache.commons.lang.StringUtils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.trino.spi.StandardErrorCode.EXCEEDED_FUNCTION_MEMORY_LIMIT;
+import static io.trino.spi.block.PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestListagg
+{
+    private QueryAssertions assertions;
+
+    @BeforeClass
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testListaggQueryWithOneValue()
+    {
+        assertThat(assertions.query(
+                "SELECT listagg(value, ',') WITHIN GROUP (ORDER BY value) " +
+                        "FROM (VALUES 'a') t(value)"))
+                .matches("VALUES (VARCHAR 'a')");
+    }
+
+    @Test
+    public void testListaggQueryWithMultipleValues()
+    {
+        assertThat(assertions.query(
+                "SELECT listagg(value, ',') WITHIN GROUP (ORDER BY value) " +
+                        "FROM (VALUES 'a', 'bb', 'ccc', 'dddd') t(value) "))
+                .matches("VALUES (VARCHAR 'a,bb,ccc,dddd')");
+    }
+
+    @Test
+    public void testListaggQueryWithImplicitSeparator()
+    {
+        assertThat(assertions.query(
+                "SELECT listagg(value) WITHIN GROUP (ORDER BY 1) " +
+                        "FROM (VALUES 'a', 'b', 'c') t(value) "))
+                .matches("VALUES (VARCHAR 'abc')");
+    }
+
+    @Test
+    public void testListaggQueryWithMultipleValuesOrderedDescending()
+    {
+        assertThat(assertions.query(
+                "SELECT listagg(value, ',') WITHIN GROUP (ORDER BY value DESC) " +
+                        "FROM (VALUES 'a', 'bb', 'ccc', 'dddd') t(value) "))
+                .matches("VALUES (VARCHAR 'dddd,ccc,bb,a')");
+    }
+
+    @Test
+    public void testListaggQueryWithMultipleValuesMultipleSortItems()
+    {
+        assertThat(assertions.query(
+                "SELECT listagg(value, ',') WITHIN GROUP (ORDER BY sortitem1, sortitem2) " +
+                        "FROM (VALUES (1, 'A', 'a') , (2, 'B', 'bb'), (2, 'C', 'ccc'), (3, 'D', 'dddd')) t(sortitem1, sortitem2, value) "))
```

Comment:
List the items in a different order to rule out the case where the test passes just because the values are scanned and produce in the order they are listed.